common/src/main/java/com/gregtechceu/gtceu/integration/GTOreProcessingWidget.java

```diff
@@ -0,0 +1,510 @@
+        addWidget(chippedSlot);
+    }
+
+    //Slots
+
+    private void addMachineSlots(List<List<ItemStack>> machines) {
+        List<ItemStack> furnaceSlot = Collections.singletonList(new ItemStack(Blocks.FURNACE.asItem()));
+        machines.add(furnaceSlot);
+        List<ItemStack> maceratorSlot = Collections.singletonList(MACERATOR[GTValues.LV].asStack());
+        machines.add(maceratorSlot);
+        List<ItemStack> washerSlot = Collections.singletonList(ORE_WASHER[GTValues.LV].asStack());
+        machines.add(washerSlot);
+        List<ItemStack> bathSlot = Collections.singletonList(CHEMICAL_BATH[GTValues.LV].asStack());
+        machines.add(bathSlot);
+        List<ItemStack> centrifugeSlot = Collections.singletonList(CENTRIFUGE[GTValues.LV].asStack());
+        machines.add(centrifugeSlot);
+        List<ItemStack> thermalCentrifugeSlot = Collections.singletonList(THERMAL_CENTRIFUGE[GTValues.LV].asStack());
+        machines.add(thermalCentrifugeSlot);
+        List<ItemStack> sifterSlot = Collections.singletonList(SIFTER[GTValues.LV].asStack());
+        machines.add(sifterSlot);
+        List<ItemStack> separatorSlot = Collections.singletonList(ELECTROMAGNETIC_SEPARATOR[GTValues.LV].asStack());
+        machines.add(separatorSlot);
+        List<ItemStack> cauldronWasherSlot = new ArrayList<>();
+        cauldronWasherSlot.add(new ItemStack(Items.CAULDRON));
+        cauldronWasherSlot.add(ORE_WASHER[GTValues.LV].asStack());
+        machines.add(cauldronWasherSlot);
+    }
+
+    private void addItemSlots(List<List<ItemStack>> mainproducts, List<List<ItemStack>> byproducts, Material material, OreProperty prop, List<Content> chanceContent) {
+        setupMainProducts(mainproducts, material, prop, chanceContent);
+        setupByproducts(byproducts, material, prop, chanceContent);
+    }
+
+    private void setupMainProducts(List<List<ItemStack>> mainproducts, Material material, OreProperty prop, List<Content> chanceContent) {
+        //Ore
+        List<ItemStack> oreSlot = Collections.singletonList(ChemicalHelper.get(ore, material));
+        mainproducts.add(oreSlot);
+        //Direct Smelt Result
+        Material smeltingResult = prop.getDirectSmeltResult() != null ? prop.getDirectSmeltResult() : material;
+        List<ItemStack> smeltSlot = new ArrayList<>();
+        if(smeltingResult.hasProperty(PropertyKey.INGOT)){
+            smeltSlot.add(ChemicalHelper.get(ingot,smeltingResult));
+        }else if(smeltingResult.hasProperty(PropertyKey.GEM)) {
+            smeltSlot.add(ChemicalHelper.get(gem,smeltingResult));
+        }else if(smeltingResult.hasProperty(PropertyKey.DUST)){
+            smeltSlot.add(ChemicalHelper.get(dust,smeltingResult));
+        }else{
+            smeltSlot.add(new ItemStack(Items.AIR));
+        }
+        mainproducts.add(smeltSlot);
+        //Crushed Ore
+        List<ItemStack> crushedSlot = Collections.singletonList(ChemicalHelper.get(crushed, material, 2 * prop.getOreMultiplier()));
```

Comment:
this could probably be like: `BuiltInRegistries.ITEM.getOrCreateTag(ChemicalHelper.getTag(TagPrefix.crushed, this)).stream().map(Holder::value).map(item -> new ItemStack(item, 2 * prop.getOreMultiplier())).toList()` to get all possible entries in the tag?

Reply:
I'll check that later today.